src/mlpack/methods/ann/layer/lp_pooling.hpp

```diff
@@ -0,0 +1,284 @@
+
+  //! Get the delta.
+  OutputDataType const& Delta() const { return delta; }
+  //! Modify the delta.
+  OutputDataType& Delta() { return delta; }
+
+  //! Get the intput width.
+  size_t const& InputWidth() const { return inputWidth; }
+  //! Modify the input width.
+  size_t& InputWidth() { return inputWidth; }
+
+  //! Get the input height.
+  size_t const& InputHeight() const { return inputHeight; }
+  //! Modify the input height.
+  size_t& InputHeight() { return inputHeight; }
+
+  //! Get the output width.
+  size_t const& OutputWidth() const { return outputWidth; }
+  //! Modify the output width.
+  size_t& OutputWidth() { return outputWidth; }
+
+  //! Get the output height.
+  size_t const& OutputHeight() const { return outputHeight; }
+  //! Modify the output height.
+  size_t& OutputHeight() { return outputHeight; }
+
+  //! Get the input size.
+  size_t InputSize() const { return inSize; }
+
+  //! Get the output size.
+  size_t OutputSize() const { return outSize; }
+
+  //! Get the normType.
+  size_t NormType() const { return normType; }
+  //! Modify the normType.
+  size_t& NormType() { return normType; }
+
+  //! Get the kernel width.
+  size_t KernelWidth() const { return kernelWidth; }
+  //! Modify the kernel width.
+  size_t& KernelWidth() { return kernelWidth; }
+
+  //! Get the kernel height.
+  size_t KernelHeight() const { return kernelHeight; }
+  //! Modify the kernel height.
+  size_t& KernelHeight() { return kernelHeight; }
+
+  //! Get the stride width.
+  size_t StrideWidth() const { return strideWidth; }
+  //! Modify the stride width.
+  size_t& StrideWidth() { return strideWidth; }
+
+  //! Get the stride height.
+  size_t StrideHeight() const { return strideHeight; }
+  //! Modify the stride height.
+  size_t& StrideHeight() { return strideHeight; }
+
+  //! Get the value of the rounding operation
+  bool const& Floor() const { return floor; }
+  //! Modify the value of the rounding operation
+  bool& Floor() { return floor; }
+
+  //! Get the size of the weights.
+  size_t WeightSize() const { return 0; }
+
+  /**
+   * Serialize the layer.
+   */
+  template<typename Archive>
+  void serialize(Archive& ar, const uint32_t /* version */);
+
+ private:
+  /**
+   * Apply pooling to the input and store the results.
+   *
+   * @param input The input to be apply the pooling rule.
+   * @param output The pooled result.
+   */
+  template<typename eT>
+  void Pooling(const arma::Mat<eT>& input, arma::Mat<eT>& output)
+  {
+    for (size_t j = 0, colidx = 0; j < output.n_cols;
+         ++j, colidx += strideHeight)
+    {
+      for (size_t i = 0, rowidx = 0; i < output.n_rows;
+           ++i, rowidx += strideWidth)
+      {
+        arma::mat subInput = input(
+            arma::span(rowidx, rowidx + kernelWidth - 1 - offset),
+            arma::span(colidx, colidx + kernelHeight - 1 - offset));
+
+        output(i, j) = pow(arma::accu(arma::pow(subInput,
+            normType)), 1.0/normType);
```

Comment:
```suggestion
            normType)), 1.0 / normType);
```

Hey, small style fix. We generally use space between operands and operators. Thanks.